api/v1alpha1/application.go

```diff
@@ -189,3 +222,40 @@ func max[T constraints.Ordered](a, b T) T {
 		return b
 	}
 }
+
+func (a *Application) LimitToCoreResourceList(resList LimitResourceList) corev1.ResourceList {
+	resourceList := make(corev1.ResourceList)
+
+	if !resList.CpuLimit.IsZero() {
+		resourceList["cpu"] = resList.CpuLimit
+	}
+
+	resourceList["memory"] = resList.MemoryLimit
+
+	if !resList.StorageLimit.IsZero() {
+		resourceList["storage"] = resList.StorageLimit
+	}
+
+	if !resList.EphemeralStorageLimit.IsZero() {
+		resourceList["ephemeral-storage"] = resList.EphemeralStorageLimit
+	}
+
+	return resourceList
+}
+
+func (a *Application) RequestToCoreResourceList(resList RequestsResourceList) corev1.ResourceList {
+	resourceList := make(corev1.ResourceList)
+
+	resourceList["cpu"] = resList.CpuRequest
+	resourceList["memory"] = resList.MemoryRequest
+
+	if !resList.StorageRequest.IsZero() {
+		resourceList["storage"] = resList.StorageRequest
+	}
+
+	if !resList.EphemeralStorageRequest.IsZero() {
+		resourceList["ephemeral-storage"] = resList.EphemeralStorageRequest
+	}
+
+	return resourceList
+}
```

Comment:
This feels like a dumb solution, but I wasn't sure if it would be possible to use the built in kubebuilder validation if I didn't make it two different structs.